### PR TITLE
Remove -test.* special case

### PR DESCRIPTION
**Pull request by dnephin**
Reverts cb88ea77998c3f024757528e3305022ab50b43be

I am not able to find any context about this change. As best as I can tell this change was made to mimic the 'go test' command.

It is breaking my use of spf13/pflag in [gotestsum](https://github.com/gotestyourself/gotestsum) by silently dropping any flag which starts with `-test.` instead giving an error.

Does anyone know why this change was made? Why does it silently drop all of these flags?

### CODE REVIEW

**Comment by fastcat** [comment]
Do flags like these show up in the args when running `go test -something`?

**Comment by dnephin** [comment]
I'm not sure what you mean. This logic only removes the flags from the list of args, it does not add any flags to the usage or help text.

**Comment by fastcat** [comment]
I'm suspecting that, if you are writing tests for pflags or code using pflags, and you run it with `go test` with some arguments (e.g. things like `-race`), values related to those arguments might show up in the args slice pflags sees, and this is intended to remove them? This is what the commit message implies.

Other PRs/issues that shed some more light on this: #168 ⇒ #169 ⇒ https://github.com/spf13/cobra/issues/597

Basically, it looks like this code is intending to ignore flags that are meant for the go runtime, not for the app itself

**Comment by cornfeedhobo** [comment]
@fastcat Great sleuthing! I'm not sure if the go test cli interface changed or what, but I can't find any more reasons to use `-test.*` flags, can you? The example in #168 is covered by simply dropping the `-test.` portion.

@dnephin After spending weeks on new regression tests, I wondered the exact same thing. I removed it and can't find any impact on my workflow.

**Comment by cornfeedhobo** [comment]
Update: I did find a case where `test.run` was required, but having removed this on my fork, I can say I haven't tripped up on this since removing it.

**Comment by fastcat** [comment]
With gotestyourself/gotestsum#176 I have a better idea of why hiding `test.*` flags is problematic -- what if you, like `gotestsum`, are trying to build what amounts to a frontend for `go test`, and thus want to accept its flags to pass them on?

A proposed middle ground here: Add an `IncludeTestFlags bool` member to `FlagSet` or `ParseErrorsWhitelist`, so that special case programs like `gotestsum` can opt in to processing `test.*` flags as needed, but "normal" programs have them hidden as they are now, on the assumption that they are meant for the test harness, not the code under test?

It seems like, to make this fully "nice", however, a different special case might be needed, specifically to treat `-test.*` as a long arg (as if it was `--test.*`) instead of as a short arg with value (`-t est.*`)?
